src/router/router.ts

```diff
@@ -0,0 +1,80 @@
+import { NameRoute } from '@/enums/name-route';
+import { Route } from '@/interfaces/route';
+import { BaseComponent } from '@/templates/base-component';
+import { routes } from './routes';
+
+export class Router {
+  static pageContainer: HTMLElement;
+  private static crntRoute: Promise<BaseComponent> | void;
+  private static routes: Route[] = routes;
+
+  private static onPathChange: (
+    route: Route,
+    props: Record<string, string>,
+  ) => Promise<BaseComponent> | void = (route, props) => {
+    if (route) {
+      return route.component(props).then((component) => {
+        this.pageContainer.append(component.getNode());
+
+        return component;
+      });
+    }
+  };
+
+  private static pathToRegex = (path: NameRoute) =>
+    new RegExp('^' + path.replace(/\//g, '\\/').replace(/:\w+/g, '([^/]+)') + '$');
+
+  private static getParams = (path: NameRoute, values: string[]) => {
+    const keys = Array.from(path.matchAll(/:(\w+)/g)).map((result) => result[1]);
+
+    return Object.fromEntries(keys.map((key, i) => [key, values[i]]));
+  };
+
+  private static defaultComponent = async (props: Record<string, string>) => {
```

Comment:
Maybe we should put it to the other route declarations and don't repeat code two times?
Also we can pass default/error routes in router config in cosntructor.